### DISCUSSION TITLE
Limit the number of devices per client.

### DISCUSSION THREAD

**Discussion by DonVietnam** [XTLS/Xray-core#3739] [Q&A] [answered]
Good afternoon, I constantly meet commercial offers that are obviously based on xray-core, but in each such offer it is stated that only 2-3 devices per user are supported, allegedly when connecting several devices exceeding the number of VPN stops working and blocks the client. But I have not found such an implementation anywhere and, in principle, the ability to somehow track the mac addresses of clients, I also empirically tried to connect to VPNs of similar providers from several devices and did not receive any blocking, is there in principle an opportunity to limit one client by the number of devices or only by traffic and about time?

**Comment by iambabyninja** [comment, marked as answer]
Good afternoon!

Due to the specifics of proxy protocols, currently, the only parameter that can at least indirectly determine the presence of multiple client connections under a single UUID is the client IP address from server logs 
Most solutions for connection limitations are based on interacting with it.

**Reply by DonVietnam** [reply]
Thank you, given the specifics of the work between Internet service providers and their customers, it is unlikely that commercial solutions will resort to such restrictions, I think it's still mostly a bluff. Moreover, empirical experience shows the same results.

**Reply by iambabyninja** [reply]
I’m not familiar with the services you’re referring to; I can only speak about my own service and the ones I’ve seen. 

In some panels, such as 3x-ui, this functionality is built-in by default and is based on fail2ban. In Marzban, for instance, there are numerous third-party solutions based on iptables/ufw or by completely disabling the abused account. 

These solutions are not perfect, but they do exist.

**Reply by KGorham123** [reply]
It is technically feasible to limit the number of device connections.
